core/src/utils/mod.rs

```diff
@@ -39,6 +42,21 @@ impl ToElements for Vec<u64> {
     }
 }
 
+// TO WORD
+// ================================================================================================
+
+/// Converts the digest obtained from the [`Blake3_256::hash`] procedure into the [`Word`].
+pub fn hash_string_to_word<'a>(value: impl Into<&'a str>) -> Word {
```

Comment:
nit: I would write this as something like: "Hashes the provided string using the BLAKE3 hash function and converts the resulting digest into a [`Word`]."

Also, can we update the `error_code_from_msg()` function to use this function under the hood?